featuretools/entityset/entityset.py

```diff
@@ -1190,43 +1190,37 @@ def add_interesting_values(self, max_values=5, verbose=False, dataframe_name=Non
             dataframes = [self[dataframe_name]]
         else:
             dataframes = self.dataframes
+
+        def add_value(df, col, val, verbose):
+            if verbose:
+                msg = "Column {}: Marking {} as an "
+                msg += "interesting value"
+                logger.info(msg.format(col, val))
+            interesting_vals = df.ww.columns[col].metadata.get('interesting_values', [])
+            df.ww.columns[col].metadata['interesting_values'] = interesting_vals + [val]
+
         for df in dataframes:
-            for column in df.columns:
-                # some heuristics to find basic 'where'-able columns
-                # include categorical columns, exclude index or foreign key columns
-                col_schema = df.ww.columns[column]
-                col_is_valid = (col_schema.is_categorical and
-                                not {'index', 'foreign_key'}.intersection(col_schema.semantic_tags))
-
-                if col_is_valid:
-                    counts = df[column].value_counts()
-
-                    # find how many of each unique value there are; sort by count,
-                    # and add interesting values to each column
-                    total_count = np.sum(counts)
-                    counts_idx = counts.index.tolist()
-                    for i in range(min(max_values, len(counts.index))):
-                        idx = counts_idx[i]
-
-                        if len(counts.index) < 25:
-                            if verbose:
-                                msg = "Column {}: Marking {} as an "
-                                msg += "interesting value"
-                                logger.info(msg.format(column, idx))
-                            interesting_vals = df.ww.columns[column].metadata.get('interesting_values', [])
-                            df.ww.columns[column].metadata['interesting_values'] = interesting_vals + [idx]
+            value_counts = df.ww.value_counts(top_n=max(25, max_values), dropna=True)
+            total_count = len(df)
 
+            for col, counts in value_counts.items():
+                if {'index', 'foreign_key'}.intersection(df.ww.semantic_tags[col]):
+                    continue
+
+                for i in range(min(max_values, len(counts))):
+                    if len(counts) < 25:
+                        # Categorical columns will include counts of 0 for all values
+                        # in categories. Only include values that have nonzero counts.
+                        if counts[i]['count'] > 0:
```

Comment:
can we move the check for count_value > 0 to before the if statement and break out of the loop once we encounter a count of 0?